File: alice/paskills/dialogovo/src/main/java/ru/yandex/alice/paskill/dialogovo/utils/AnythingToStringJacksonDeserializer.java
package ru.yandex.alice.paskill.dialogovo.utils;

import java.io.IOException;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.core.TreeNode;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.JsonDeserializer;

public class AnythingToStringJacksonDeserializer extends JsonDeserializer<String> {

    @Override
    public String deserialize(JsonParser jp, DeserializationContext ctxt) throws IOException {
        TreeNode tree = jp.getCodec().readTree(jp);
        return tree.toString();
    }
}

File: alice/kronstadt/core/src/main/java/utils/AnythingToStringJacksonDeserializer.kt
package ru.yandex.alice.kronstadt.core.utils

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.core.TreeNode
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.JsonDeserializer
import java.io.IOException

class AnythingToStringJacksonDeserializer : JsonDeserializer<String>() {

    @Throws(IOException::class)
    override fun deserialize(jp: JsonParser, ctxt: DeserializationContext): String {
        val tree = jp.codec.readTree<TreeNode>(jp)
        return tree.toString()
    }
}
